server/src/main/java/org/opensearch/index/mapper/ObjectMapper.java

```diff
@@ -606,7 +610,40 @@ protected static void parseProperties(ObjectMapper.Builder objBuilder, Map<Strin
                 parserContext.indexVersionCreated(),
                 "DocType mapping definition has unsupported parameters: "
             );
+        }
+
+        /**
+         * Validates whether the specified field type is compatible with Parquet data format.
+         *
+         * @param fieldType the field type to validate
+         * @param fieldName the name of the field being validated
+         * @throws MapperParsingException if the field type is not compatible with Parquet format
+         */
+        private static void validateParquetDataFormatCompatibility(String fieldType, String fieldName) {
+            if (!isParquetCompatibleType(fieldType)) {
+                throw new MapperParsingException(
+                    String.format("Field type '%s' for field '%s' is not compatible with Parquet data format. " +
+                            "Allowed types for Parquet: %s",
+                        fieldType, fieldName, getAllowedParquetTypes())
+                );
+            }
+        }
 
+        /**
+         * Checks if the given field type is compatible with Parquet data format.
+         */
+        private static boolean isParquetCompatibleType(String fieldType) {
+            return getAllowedParquetTypes().contains(fieldType);
+        }
+
+        /**
+         * Returns the set of field types that are allowed for Parquet data format.
+         */
+        private static Set<String> getAllowedParquetTypes() {
```

Comment:
This should be coming from the data format plugin's implementation. We already have a method there. Could you see if you can extend the implementation there?